zenpacklib.py

```diff
@@ -584,13 +584,16 @@ def _create_catalog(self, name, scope='device'):
         return zcatalog
 
     @classmethod
-    def _create_indexes(cls, zcatalog, spec):
+    def _create_indexes(cls, zcatalog, spec, leaveObjects=True):
         from Products.ZCatalog.Catalog import CatalogError
         from Products.Zuul.interfaces import ICatalogTool
         catalog = zcatalog._catalog
 
-        classname = spec.get(
-            'class', 'Products.ZenModel.DeviceComponent.DeviceComponent')
+        # I think this is the original intent for setting classname, not sure why it would fail
+        try:
+            classname = '%s.%s' % (cls.__module__, cls.__class__.__name__)
```

Comment:
Agreed -- I wrote it this way because like I said I'm guessing at the original intent